jobs/semantic-release/action.yaml

```
@@ -0,0 +1,83 @@
+name: semantic-release
+description: Release a new version of the project using conventional commits
+author: Lenra-io
+branding:
+  color: orange
+  icon: tag
+
+inputs:
+  node-version:
+    description: The version of node to use
+    required: false
+    default: '18'
+  dry-run:
+    description: Whether to publish the release or not
+    required: false
+    default: 'false'
+  release-branches:
+    description: The branches on which releases should happen, comma separated name of branches
+    required: false
+    default: 'main'
+  prerelease-branches:
+    description: The branches on which prereleases should happen, comma separated name of branches
+    required: false
+    default: 'rc,beta,alpha'
+  assets:
+    description: 'The assets to upload to the release, comma separated list of `{"path": "<glob>", "name": "<name>"}` objects'
+    required: false
+    default: ''
+
+outputs:
+  release-published:
+    description: "Whether a release was published or not"
+    value: ${{ steps.semantic-release.outputs.new-release-published }}
+  release-version:
+    description: "The next version discovered by conventional commits"
+    value: ${{ steps.semantic-release.outputs.new-release-version }}
+
+runs:
+  using: "composite"
+  steps:
```

Check if branch name matches the release or prerelease branches ?

---

Did you think something like this is Ok ? 
```suggestion
  if: ${{ contains(format('{0},{1}', inputs.release-branches, inputs.prerelease-branches), github.ref_name) }}
  steps:
```

The `contains` here look if the input string contain the string of the current branch name.
For our usecase I think it's Ok, but for more global use it can cause issues of similar branch name.